openssl-static/pom.xml

```diff
@@ -0,0 +1,206 @@
+                <gunzip src="${project.build.directory}/openssl-${opensslVersion}.tar.gz"
+                        dest="${project.build.directory}/" />
+                <untar src="${project.build.directory}/openssl-${opensslVersion}.tar"
+                       dest="${project.build.directory}/" />
+                <!-- The untar task does not preserve permissions. Make everything executable. -->
+                <chmod perm="ug+wx" type="file">
+                  <fileset dir="${opensslBuildDir}/" includes="**/*" />
+                </chmod>
+              </target>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+
+      <!-- Configure the distribution statically linked against OpenSSL and APR -->
+      <plugin>
+        <groupId>org.fusesource.hawtjni</groupId>
+        <artifactId>maven-hawtjni-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>build-native-lib</id>
+            <goals>
+              <goal>generate</goal>
+              <goal>build</goal>
+            </goals>
+            <phase>compile</phase>
+            <configuration>
+              <name>netty-tcnative</name>
+              <nativeSourceDirectory>${project.basedir}/src/main/c</nativeSourceDirectory>
+              <libDirectory>${nativeLibOnlyDir}</libDirectory>
+              <forceAutogen>${forceAutogen}</forceAutogen>
+              <forceConfigure>${forceConfigure}</forceConfigure>
+              <windowsBuildTool>msbuild</windowsBuildTool>
+              <configureArgs>
+                <configureArg>--with-ssl=${sslHome}</configureArg>
+                <configureArg>--with-apr=${aprHome}</configureArg>
+              </configureArgs>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+    </plugins>
+  </build>
+
+  <profiles>
+    <profile>
+      <id>build-openssl-windows</id>
+      <activation>
+        <os>
+          <family>windows</family>
+        </os>
+      </activation>
+      <build>
+        <plugins>
+          <plugin>
+            <artifactId>maven-antrun-plugin</artifactId>
+            <executions>
+              <execution>
+                <id>build-openssl</id>
+                <phase>generate-sources</phase>
+                <goals>
+                  <goal>run</goal>
+                </goals>
+                <configuration>
+                  <target>
+                    <!-- Add the ant tasks from ant-contrib -->
+                    <taskdef resource="net/sf/antcontrib/antcontrib.properties" />
+
+                    <!-- Build for the correct platform -->
+                    <pathconvert property="sslHomePath" targetos="windows">
+                      <path location="${sslHome}"/>
+                    </pathconvert>
+                    <if>
+                      <equals arg1="${archBits}" arg2="32" />
+                      <then>
+                        <echo message="Building OpenSSL for Win32" />
```

Comment:
we do by default ... see the else.  `archBits` defaults to 64, but can be overridden if we *really* want to.